src/qibolab/pulses/sequence.py

```diff
@@ -34,8 +34,13 @@ def channel_duration(self, channel: str) -> float:
         """Duration of the given channel."""
         return sum(item.duration for item in self[channel])
 
-    def __add__(self, other: "PulseSequence") -> "PulseSequence":
-        """Create a PulseSequence which is self + necessary delays to
-        synchronize channels + other."""
-        # TODO: implement
-        ...
+    def append(self, other: "PulseSequence") -> None:
```

Comment:
Why did you decide to avoid the dunder?

In any case, if you want a name, I'd suggest:
```suggestion
    def extend(self, other: "PulseSequence") -> None:
```
because the syntax (and semantic) of `list.append()` is `list -> element -> list`, while `list.extend()` is `list -> list -> list` (ok, not returning, but mutating in-place, but the meaning is that).

Reply:
This has been renamed to extend. The reason I decided not to go with the dunder method is that `PulseSequence` object is by design a highly mutable object, and main use case would be to modify it in place anyways, so ps2 = ps0 + ps1 is not predicted to be a common use case

Reply:
But if `__add__` is present, you can always use as:
```py
ps0 += ps1
```
which is basically `.extend()`.

Reply:
You can also use it as c = a + b, while with extend you can't, which is more clear interface.

Reply:
Well, `c = a + b` is the exact same interface of lists, as also `.extend()` is. Since you're mimicking a list, it doesn't sound unreasonable to support its basic operations.

You could even implement `.__add__()` and `.__iadd__()` on top of `.extend()`.